src/BuildPrediction/ProjectGraphPredictionExecutor.cs

```diff
@@ -0,0 +1,152 @@
+            : this(projectGraphPredictors, projectPredictors, null)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="ProjectGraphPredictionExecutor"/> class.</summary>
+        /// <param name="projectGraphPredictors">The set of <see cref="IProjectGraphPredictor"/> instances to use for prediction.</param>
+        /// <param name="projectPredictors">The set of <see cref="IProjectPredictor"/> instances to use for prediction.</param>
+        /// <param name="options">The options to use for prediction.</param>
+        public ProjectGraphPredictionExecutor(
+            IEnumerable<IProjectGraphPredictor> projectGraphPredictors,
+            IEnumerable<IProjectPredictor> projectPredictors,
+            ProjectPredictionOptions options)
+        {
+            _projectGraphPredictors = projectGraphPredictors
+                .ThrowIfNull(nameof(projectGraphPredictors))
+                .Select(p => new ValueAndTypeName<IProjectGraphPredictor>(p))
+                .ToArray(); // Array = faster parallel performance.
+            _projectPredictors = projectPredictors
+                .ThrowIfNull(nameof(projectPredictors))
+                .Select(p => new ValueAndTypeName<IProjectPredictor>(p))
+                .ToArray(); // Array = faster parallel performance.
+            _options = options ?? new ProjectPredictionOptions();
+        }
+
+        /// <summary>
+        /// Constructs a graph starting from the given graph entry points and execute all project and
+        /// graph predictors against the resulting projects.
+        /// </summary>
+        /// <param name="graphEntryPoints">The entry points to use for the graph creation.</param>
+        /// <param name="projectCollection">The collection with which all projects in the graph should be associated.</param>
+        /// <returns>An object describing all predicted inputs and outputs.</returns>
+        public ProjectPredictions PredictInputsAndOutputs(
+            IEnumerable<ProjectGraphEntryPoint> graphEntryPoints,
+            ProjectCollection projectCollection)
+        {
+            var projectPredictionCollector = new DefaultProjectPredictionCollector();
+            PredictInputsAndOutputs(graphEntryPoints, projectCollection, projectPredictionCollector);
+            return projectPredictionCollector.Predictions;
+        }
+
+        /// <summary>
+        /// Constructs a graph starting from the given graph entry points and execute all project and
+        /// graph predictors against the resulting projects.
+        /// </summary>
+        /// <param name="graphEntryPoints">The entry points to use for the graph creation.</param>
+        /// <param name="projectCollection">The collection with which all projects in the graph should be associated.</param>
+        /// <param name="projectPredictionCollector">The prediction collector to use.</param>
+        public void PredictInputsAndOutputs(
```

Comment:
This probably needs to return the resulting `ProjectGraph`. Or maybe it should take it as an input instead of the entry points instead?

Reply:
Fixed by having this take a ProjectGraph instead of constructing one.